core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java

```diff
@@ -366,19 +364,22 @@ public EmailMessage next() {
 		}
 	}
 
-	
+
 	@Override
 	public boolean folderExists(String folder) throws FileSystemException {
-		FolderId folderId;
+		ExchangeFileSystemResolver resolver = getResolver(folder);
+		registerMailbox(resolver.getMailbox());
+
+		FolderId folderId = null;
 		try {
-			folderId = findFolder(folder);
+			folderId = findFolder(cache.getBaseFolderId(resolver.getMailbox()), resolver.getFolderName());
```

Comment:
Waarom gebruik je hier `findFolder()`, en op andere plekken `getFolderIdByFolderName()`? 
Zou je dat kunnen uniformeren? Dus hier ook `getFolderIdByFolderName()`?

Reply:
De reden hiervoor was dat findFolder altijd de folder opvroeg van de service en geen gebruikt maakt van de cache.

Ik heb getFolderIdByFolderName aangepast om een extra argument te accepteren waarmee de aanroeper kan controleren of de cache gebruikt mag worden.